libs/service-portal/core/src/screens/NoDataScreen/NoDataScreen.tsx

```diff
@@ -7,98 +7,139 @@ import {
   TagVariant,
   Text,
   Button,
+  IconProps,
 } from '@island.is/island-ui/core'
 import * as styles from '../ErrorScreen/ErrorScreen.css'
 import { Link } from 'react-router-dom'
 import { ServicePortalPath } from '../../lib/navigation/paths'
 
+interface ButtonType {
+  link?: ServicePortalPath | string
+  onClick?: () => void
+  icon?: Pick<IconProps, 'icon' | 'type'>
+  type: 'internal' | 'external' | 'click'
```

Comment:
This is a lot of boilerplate for common logic. Would be nice to upgrade the design system so buttons and links can support all of this automatically.

Reply:
Yeah well Im getting different comments from the UI team regarding what the purpose of this UI library is. Some say it should only be a skeleton and the developer should handle more detailed actions and UI, while other want to refactor to handle all possible versions 🤷‍♀️ So I'm not gonna refactor this as well 😅